File: com/pdfjet/Legal.java
/**
 * Legal.java
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
package com.pdfjet;

/**
 * Used to specify PDF page with size <strong>Legal</strong>.
 * For more information about the page size classes - A3, A4, A5, B5, Executive, Letter, Legal and Tabloid - see the Page class.
 */
public class Legal {
    /** Default Constructor */
    public Legal() {
    }

    /** Portrait orientation */
    public static final float[] PORTRAIT = new float[] {612.0f, 1008.0f};
    /** Landscape orientation */
    public static final float[] LANDSCAPE = new float[] {1008.0f, 612.0f};
}

File: src/legal/legal.go
// legal.go
//
// Copyright (c) 2026 PDFjet Software
// Licensed under the MIT License. See LICENSE file in the project root.

// Package legal defines the legal page size in points.
package legal

import "github.com/edragoev1/pdfjet/v9/src/pagesize"

// Portrait returns the legal page size in portrait orientation.
func Portrait() pagesize.PageSize {
	return pagesize.NewPageSize(612.0, 1008.0)
}

// Landscape returns the legal page size in landscape orientation.
func Landscape() pagesize.PageSize {
	return pagesize.NewPageSize(1008.0, 612.0)
}
